addons/sale/models/sale_order_line.py

```diff
@@ -925,6 +928,16 @@ def _onchange_product_packaging_id(self):
                     },
                 }
 
+    @api.onchange('price_unit')
+    def _onchange_price_unit(self):
+        if self.price_unit and self.price_unit != self.product_id.lst_price:
+            self.is_price_set_manually = True
+
+    @api.onchange('pricelist_item_id')
+    def _onchange_pricelist_item_id(self):
+        if self.pricelist_item_id:
+            self.is_price_set_manually = False
```

Comment:
This doesn't seems right 
can you explain why we don't need to set is_price_set_manually to false when not pricelist is apply?

Reply:
By default, when no pricelist is applied, is_price_set_manually is False. If the price is manually changed, it becomes True. If a pricelist is later applied with a corresponding pricelist_item_id, is_price_set_manually should revert to False to adopt the pricelist price.

Reply:
But shouldn't we should be consistent? :thinking: 
Like:
E.G
Product Test price 100
Price List:
5 qty price 90
10 qty price 80

Now I add that line with 5 qty and update its price to 85
If I change it to < 5 then it keep 85 value
It I change it to > 10 then it update price to 80

so I was thinking shouldn't it be consistent
If I change qty less then 5 then it should set Product value 100
OR
If I change my qty to greater then 10 then it should keep 85